src/setup/mod.rs

```diff
@@ -247,32 +247,17 @@ fn parse_version(version_str: &str) -> Option<(u32, u32)> {
 #[instrument(level = "trace", skip_all)]
 fn check_rust_toolchains_targets() -> Result<Vec<Dependency>> {
     let mut missing_deps = Vec::new();
-    let output = Command::new("rustup").arg("show").output()?.stdout;
-    let output = String::from_utf8_lossy(&output);
 
-    let original_default = output.split('\n').fold("", |d, item| {
-        if !item.contains("(default)") {
-            d
-        } else {
-            item.split(' ').nth(0).unwrap_or("")
-        }
-    });
+    let has_wasm32_wasi = Command::new("sh")
+        .arg("-c")
+        .arg("rustup show | grep -A 255 'installed targets' | grep -q 'wasm32-wasip1'")
+        .output()?
+        .status;
```

Comment:
This is nice and clean, but I'm nervous that we are introducing an edge case with the `-A 255`.

How about instead we do something like:
```
    let output = Command::new("rustup").arg("+stable").arg("show").output()?.stdout;
    let output = String::from_utf8_lossy(&output);

    let has_wasm32_wasi = output
        .split('\n')
        .fold(false, |acc, item| acc || item == "wasm32-wasip1");
```

which doesn't have any edgecases, and also forces `+stable` (which I've realized we should add in other `cargo` calls -- but that can be for another PR)

Reply:
Eh, I'll just do it while I'm here.